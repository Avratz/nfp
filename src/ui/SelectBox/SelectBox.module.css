.selectBox {
  position: relative;
}

.button,
.options {
  display: flex;
  border: 2px solid var(--neutral-3);
  width: 100%;
  border-radius: 0.75rem;
  cursor: pointer;
  font-family: var(--font-poppins);
  font-weight: var(--font-medium);
  font-size: var(--text-sm);
  color: var(--neutral-8);
}

.button {
  padding: 0.375rem;
  padding-left: 1rem;
  background-color: transparent;
  justify-content: space-between;
  align-items: center;
}

.label {
  display: block;
  text-transform: uppercase;
  margin-bottom: 0.75rem;
  font-family: var(--font-poppins);
  font-weight: var(--font-bold);
  font-size: var(--text-xs);
  color: var(--neutral-5);
}

.options {
  flex-direction: column;
  gap: 0.5rem;
  position: absolute;
  background-color: var(--neutral-2);
  list-style: none;
  z-index: 100;
  box-shadow: 0 4rem 4rem -3rem rgba(31, 47, 70, 0.12);
  padding: 0.5rem 0.375rem;
  animation: var(--fadeIn);
}

.icon {
  width: 32px;
  height: 32px;
  background-color: transparent;
  border: 2px solid var(--neutral-3);
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding-top: 1px;
  color: var(--neutral-4);
}

.option {
  padding: 0.5rem;
  border-radius: 0.5rem;
  font-weight: var(--font-regular);
  transition: background-color 0.3s;
}

.selected {
  background-color: var(--neutral-1);
}

.active {
  background-color: var(--neutral-1);
}
