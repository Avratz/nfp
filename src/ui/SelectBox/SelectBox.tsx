'use client'
import { Listbox } from '@headlessui/react'
import { cx } from 'class-variance-authority'
import { ReactNode } from 'react'
import { Icon } from '../Icon/Icon'

import styles from './SelectBox.module.css'

type SelectBoxProps = {
  placeholder: string
  selected?: { name: string }
  setSelected: (selected: { name: string }) => void
  children: ReactNode
  label?: string
  className?: string
}

export function SelectBox({
  placeholder,
  selected,
  setSelected,
  children,
  label,
  className,
}: SelectBoxProps) {
  return (
    <Listbox value={selected} onChange={setSelected}>
      <div className={styles.selectBox}>
        {label ? (
          <Listbox.Label className={styles.label}>{label}</Listbox.Label>
        ) : null}
        <Listbox.Button className={cx(styles.button, className)}>
          <span>{selected?.name || placeholder}</span>
          <span className={styles.icon}>
            <Icon type="arrowDown" size="xs" />
          </span>
        </Listbox.Button>

        <Listbox.Options className={styles.options}>{children}</Listbox.Options>
      </div>
    </Listbox>
  )
}

SelectBox.Option = Listbox.Option
