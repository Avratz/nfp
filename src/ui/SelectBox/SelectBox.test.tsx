import { render } from '@testing-library/react'
import { SelectBox, OptionBox } from './SelectBox'

jest.mock('../Icon/Icon', () => {
  function Icon() {
    return <div>u</div>
  }

  Icon.displayName = 'Icon'
  return {
    __esModule: true,
    Icon,
  }
})

describe('SelectBox', () => {
  it('renders default', () => {
    const { container } = render(
      <SelectBox placeholder="Select box" setSelected={() => null}>
        SelectBox
      </SelectBox>,
    )
    expect(container).toMatchSnapshot()
  })

  it('renders with label', () => {
    const { container } = render(
      <SelectBox
        placeholder="Select box"
        setSelected={() => null}
        label="Label"
      >
        SelectBox
      </SelectBox>,
    )
    expect(container).toMatchSnapshot()
  })
})

describe('OptionBox', () => {
  it('renders default', () => {
    const { container } = render(
      <SelectBox placeholder="Select box" setSelected={() => null}>
        <OptionBox value={{ name: 'Option', value: 'option' }}>
          Option
        </OptionBox>
      </SelectBox>,
    )
    expect(container).toMatchSnapshot()
  })
})
