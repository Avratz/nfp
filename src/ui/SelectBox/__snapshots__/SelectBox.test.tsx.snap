// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionBox renders default 1`] = `
<div>
  <div
    class="selectBox"
  >
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      class="button"
      data-headlessui-state=""
      id="headlessui-listbox-button-:r5:"
      type="button"
    >
      <span>
        Select box
      </span>
      <span
        class="icon"
      >
        <div>
          u
        </div>
      </span>
    </button>
  </div>
</div>
`;

exports[`SelectBox renders default 1`] = `
<div>
  <div
    class="selectBox"
  >
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      class="button"
      data-headlessui-state=""
      id="headlessui-listbox-button-:r0:"
      type="button"
    >
      <span>
        Select box
      </span>
      <span
        class="icon"
      >
        <div>
          u
        </div>
      </span>
    </button>
  </div>
</div>
`;

exports[`SelectBox renders with label 1`] = `
<div>
  <div
    class="selectBox"
  >
    <label
      class="label"
      data-headlessui-state=""
      id="headlessui-listbox-label-:r2:"
    >
      Label
    </label>
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="headlessui-listbox-label-:r2: headlessui-listbox-button-:r3:"
      class="button"
      data-headlessui-state=""
      id="headlessui-listbox-button-:r3:"
      type="button"
    >
      <span>
        Select box
      </span>
      <span
        class="icon"
      >
        <div>
          u
        </div>
      </span>
    </button>
  </div>
</div>
`;
